tutorials/dataframe/df106_HiggsToFourLeptons_withVariations.C

```diff
@@ -0,0 +1,279 @@
+/// \notebook -draw
+/// The Higgs to four lepton analysis from the ATLAS Open Data release of 2020, with RDataFrame.
+///
+/// This tutorial is the Higgs to four lepton analysis from the ATLAS Open Data release in 2020
+/// (http://opendata.atlas.cern/release/2020/documentation/). The data was taken with the ATLAS detector
+/// during 2016 at a center-of-mass energy of 13 TeV. The decay of the Standard Model Higgs boson
+/// to two Z bosons and subsequently to four leptons is called the "golden channel". The selection leads
+/// to a narrow invariant mass peak on top a relatively smooth and small background, revealing
+/// the Higgs at 125 GeV.
+/// The analysis is translated to an RDataFrame workflow processing about 300 MB of simulated events and data.
+///
+/// Lepton scale factors are applied to leptons in MC simulations to correct for the differences in the
+/// reconstruction, identification and trigger mechanisms used for MC samples and real data.
+/// Systematic uncertainties for those MC scale factors are evaluated and the Vary function of RDataFrame is used to
+/// scale the variations to the final four leptons mass distribution.
+///
+///
+/// \macro_code
+/// \macro_image
+/// \date March 2020
+/// \author Stefan Wunsch (KIT, CERN)
+/// \date August 2022
+/// \author Julia Mathe (CERN)
+/// \date August 2023
+/// \author Marta Czurylo (CERN)
+
+#include <Math/Vector4D.h>
+#include "ROOT/RDFHelpers.hxx"
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RVec.hxx"
+#include "TCanvas.h"
+#include "TH1D.h"
+#include "TLatex.h"
+#include <TLegend.h>
+#include "TProfile.h"
+#include "TStyle.h"
+
+using namespace ROOT::VecOps;
+using cRVecF = const ROOT::RVecF &;
+using namespace ROOT::RDF::Experimental;
+
+// Define functions needed in the analysis
+// Select events for the analysis
+bool GoodElectronsAndMuons(const ROOT::RVecI &type, cRVecF pt, cRVecF eta, cRVecF phi, cRVecF e, cRVecF trackd0pv,
+                           cRVecF tracksigd0pv, cRVecF z0)
+{
+   for (size_t i = 0; i < type.size(); i++) {
+      ROOT::Math::PtEtaPhiEVector p(pt[i] / 1000.0, eta[i], phi[i], e[i] / 1000.0);
+      if (type[i] == 11) {
+         if (pt[i] < 7000 || abs(eta[i]) > 2.47 || abs(trackd0pv[i] / tracksigd0pv[i]) > 5 ||
+             abs(z0[i] * sin(p.Theta())) > 0.5)
+            return false;
+      } else {
+         if (abs(trackd0pv[i] / tracksigd0pv[i]) > 5 || abs(z0[i] * sin(p.Theta())) > 0.5)
+            return false;
+      }
+   }
+   return true;
+}
+
+// Compute the invariant mass of a four-lepton-system.
+float ComputeInvariantMass(cRVecF pt, cRVecF eta, cRVecF phi, cRVecF e)
+{
+   ROOT::Math::PtEtaPhiEVector p1(pt[0], eta[0], phi[0], e[0]);
+   ROOT::Math::PtEtaPhiEVector p2(pt[1], eta[1], phi[1], e[1]);
+   ROOT::Math::PtEtaPhiEVector p3(pt[2], eta[2], phi[2], e[2]);
+   ROOT::Math::PtEtaPhiEVector p4(pt[3], eta[3], phi[3], e[3]);
+   return (p1 + p2 + p3 + p4).M() / 1000;
+}
+
+void df106_HiggsToFourLeptons_addVariations()
```

Comment:
```suggestion
void df106_HiggsToFourLeptons_withVariations()
```
To run this as a ROOT macro, the function has to match the filename.

By the way, why don't you just keep the old filename `df106_HiggsToFourLeptons.C/.py`? If the new tutorial with variations is meant to replace the old one, why not just keep the same name?

Reply:
true, I think I changed the filename last minute. I was not sure about naming, but thinking about this now again, I agree, let's keep the old names